src/core/shape_inference/include/interpolate_shape_inference.hpp

```diff
@@ -0,0 +1,228 @@
+            output[i] = output_value[i];
+        }
+        return true;
+    }
+    return false;
+}
+
+template <>
+inline bool get_data_as_shape<ov::PartialShape>(
+    size_t idx,
+    const ov::Node* op,
+    ov::PartialShape& output,
+    const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data) {
+    if (ov::evaluate_as_partial_shape(op->input_value(idx), output)) {
+        return true;
+    }
+    return false;
+}
+
+namespace v0 {
+template <class T>
+void shape_infer(const Interpolate* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 2 && output_shapes.size() == 1);
+
+    const auto& input_shape = input_shapes[0];
+    auto& output_shape = output_shapes[0];
+    output_shape = input_shape;
+
+    if (input_shape.rank().is_static()) {
+        auto input_rank = input_shape.size();
+        NODE_VALIDATION_CHECK(op,
+                              std::all_of(op->m_attrs.axes.begin(),
+                                          op->m_attrs.axes.end(),
+                                          [input_rank](size_t axis) {
+                                              return axis < input_rank;
+                                          }),
+                              "Axis value should less than input rank. ",
+                              "Got: input rank ",
+                              input_rank,
+                              ", axes ",
+                              op->m_attrs.axes);
+
+        T target_spatial_shape;
+        if (get_data_as_shape<T>(1, op, target_spatial_shape, constant_data)) {
+            size_t i = 0;
+            for (auto axis : op->m_attrs.axes) {
+                output_shape[axis] = target_spatial_shape[i++];
+            }
+        } else {
+            for (auto axis : op->m_attrs.axes) {
+                output_shape[axis] = ov::Dimension::dynamic();
+            }
+        }
+    }
+}
+}  // namespace v0
+
+namespace v4 {
+template <typename T>
+std::vector<T> correct_pad(const std::vector<T>& p, size_t rank) {
+    size_t pad_len = p.size();
+    if (pad_len == rank) {
+        return p;
+    }
+
+    std::vector<T> result;
+    if (pad_len > rank) {
+        result.insert(result.end(), p.begin(), p.begin() + rank);
+    } else {
+        result = p;
+        result.insert(result.end(), rank - pad_len, T{});
+    }
+    return result;
+}
+
+template <typename T>
+void correct_pads_attr(const Interpolate* op,
+                       std::vector<size_t>& pads_begin,
+                       std::vector<size_t>& pads_end,
+                       const std::vector<T>& input_shapes) {
+    auto input_shape = input_shapes[0];
+    pads_begin = op->m_attrs.pads_begin;
+    pads_end = op->m_attrs.pads_end;
```

Comment:
Can we remove this? It's not used anywhere

Reply:
It was used in [shape_infer](https://github.com/openvinotoolkit/openvino/pull/8521/files#diff-bc2e221ff7c8a03c2a233812ca3329508a8df1986f33d67db2f56da9a11083a2R178)

Reply:
I mean that we redefine these values [at the end](https://github.com/openvinotoolkit/openvino/blob/b20956f39fd83aa549affd43cbdff4557b0d9936/src/core/shape_inference/include/interpolate_shape_inference.hpp#L125) of this function.
Does it make sense to define these values here if we will redefine them later?

Reply:
You are right, will remove it.

Reply:
Removed.